bingo/symbolic_regression/atomic_potential_regression.py

```diff
@@ -52,6 +52,9 @@ def evaluate_fitness_vector(self, individual):
 
         return np.array(err_vec).flatten()
 
+    def get_jacobian(self, individual):
```

Comment:
rather than having every fitness function be required to implement this function lets implement this with mixins:

1) make a GradientMixin class which has the abstract function for get_gradient

2) make a subclass of GradientMixin  called VectorGradientMixin.  this class has the get_gradient implementation that you currently have in the vectorbasedfitnessfunction  (including the abstract get_jacobian).

3) when getting ready to call scipy in the continuous local optimization you can test whether the fitnessfunction is a GradientMixin/VectorGradientMixin  and only use the gradient based omprimization in that case.

this also helps because then we don't have to modify all the implemented fitness functions, they just dont use the gradientMixin

Reply:
sounds great, I did this in the new changes!